hack/build-image

```diff
@@ -733,15 +739,25 @@ class QMatcher:
         return False
 
 
+def _autobuild(cli, target, is_index=None):
+    if is_index is None:
+        is_index = isinstance(target, TargetIndex)
+    if cli.condition == REBUILD:
+        rebuild(cli, target)
+    elif maybe_container_id(cli, target):
+        logger.debug("target item exists: %s", target)
+    elif cli.condition == AUTO or (cli.condition == AUTO_INDEX and is_index):
+        logger.debug("target item auto build: %s", target)
+        build(cli, target)
+    else:
+        log.error("no existing image or index: %s", target)
+        raise ValueError("not present", target)
+
+
 def push(cli, target):
     """Command to push images."""
-    if cli.push_state == "rebuild":
-        build(cli, target)
-    if cli.push_state == "exists":
-        try:
-            container_id(cli, target)
-        except subprocess.CalledProcessError:
-            build(cli, target)
+    is_index = isinstance(target, TargetIndex)  # is it a manifest push?
+    _autobuild(cli, target, is_index=is_index)
```

Comment:
```suggestion
    _autobuild(cli, target)
```

The above will do, right?

Reply:
`is_index` is passed as `manifest` arg of container_push below.  Making `_autobuild` take this param was not strictly necessary but for some reason I felt compelled to not repeat the isinstance check :shrug: